autogen/agentchat/agent.py

```diff
@@ -1,51 +1,62 @@
-    """(In preview) An abstract class for AI agent.
+@runtime_checkable
+class Agent(Protocol):
+    """(In preview) A protocol for Agent.
 
     An agent can communicate with other agents and perform actions.
     Different agents can differ in what actions they perform in the `receive` method.
     """
 
-    def __init__(
-        self,
-        name: str,
-    ):
-        """
-        Args:
-            name (str): name of the agent.
-        """
-        # a dictionary of conversations, default value is list
-        self._name = name
-
     @property
-    def name(self):
-        """Get the name of the agent."""
-        return self._name
+    def name(self) -> str:
+        """The name of the agent."""
+        ...
 
-    def send(self, message: Union[Dict, str], recipient: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract method) Send a message to another agent."""
+    @property
+    def description(self) -> str:
+        """The description of the agent. Used for the agent's introduction in
+        a group chat setting."""
+        ...
 
-    async def a_send(self, message: Union[Dict, str], recipient: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract async method) Send a message to another agent."""
+    def send(
+        self,
+        message: Union[Dict[str, Any], str],
+        recipient: "Agent",
+        request_reply: Optional[bool] = None,
+    ) -> None:
+        """Send a message to another agent."""
+        ...
 
-    def receive(self, message: Union[Dict, str], sender: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract method) Receive a message from another agent."""
+    async def a_send(
+        self,
+        message: Union[Dict[str, Any], str],
+        recipient: "Agent",
+        request_reply: Optional[bool] = None,
+    ) -> None:
+        """(Async) Send a message to another agent."""
+        ...
 
-    async def a_receive(self, message: Union[Dict, str], sender: "Agent", request_reply: Optional[bool] = None):
-        """(Abstract async method) Receive a message from another agent."""
+    def receive(
+        self, message: Union[Dict[str, Any], str], sender: "Agent", request_reply: Optional[bool] = None
+    ) -> None:
+        """Receive a message from another agent."""
 
-    def reset(self):
-        """(Abstract method) Reset the agent."""
+    async def a_receive(
+        self, message: Union[Dict[str, Any], str], sender: "Agent", request_reply: Optional[bool] = None
+    ) -> None:
+        """(Async) Receive a message from another agent."""
+        ...
 
     def generate_reply(
         self,
-        messages: Optional[List[Dict]] = None,
+        messages: Optional[List[Dict[str, Any]]] = None,
         sender: Optional["Agent"] = None,
-        **kwargs,
-    ) -> Union[str, Dict, None]:
-        """(Abstract method) Generate a reply based on the received messages.
+        exclude: Optional[List[Callable[..., Any]]] = None,
```

Comment:
What's the purpose of this parameter?

Reply:
Good catch, should remove this from the interface method.

Reply:
Removed exclude from interface arguments.